src/types/Product.ts

```diff
@@ -0,0 +1,91 @@
+import { Image } from './Image';
+
+export type ProductNode = {
+  readonly description: string;
+  readonly descriptionHtml: string;
+  readonly featuredImage: Image;
+  readonly handle: string;
+  readonly hasOnlyDefaultVariant: string;
+  readonly hasOutOfStockVariants: string;
+  readonly id: string;
+  readonly images: ImagesWrapper;
+  readonly options: readonly Option[];
+  readonly priceRangeV2: PriceRange;
+  readonly productType: string;
+  readonly status: string;
+  readonly tags: string[];
+  readonly title: string;
+  readonly totalInventory: number;
+  readonly totalVariants: number;
+  readonly variants: Variants;
+  readonly vendor: string;
+};
+
+export type ImagesWrapper = {
+  readonly edges: Image[];
+  readonly pageInfo: PageInfo;
+};
+
+export type Option = {
+  readonly id: string;
+  readonly name: string;
+  readonly position: number;
+  readonly values: string[];
+};
+
+export type PriceRange = {
+  readonly maxVariantPrice: VarientPrice;
+  readonly minVariantPrice: VarientPrice;
+};
+
+export type VarientPrice = {
+  readonly amount: number;
+  readonly currencyCode: string;
+};
+
+export type Variants = {
+  readonly edges: readonly VariantEdge[];
+  readonly pageInfo: PageInfo;
+};
+
+export type VariantEdge = {
+  readonly node: readonly VariantNode[];
+  readonly sku: string;
+  readonly taxable: string;
+  readonly title: string;
+};
+
+export type VariantNode = {
```

Comment:
```suggestion
export type Variant = {
```

Please make sure all type names match with Java POJOs

Reply:
Done 👍🏾